app/assets/javascripts/file_dropzone.js.es6

```diff
@@ -7,9 +7,26 @@ window.App.FileDropzone = class FileDropzone {
   }
 
   init() {
-    // If we have a single file input on the page, initialize the zone
-    if ($("input[type='file']").length == 1) {
-      this.insert_dropzone($("input[type='file']"))
+    // If we have a single visible file input on the page, initialize the zone
+    if ($("input[type='file']:visible").length == 1) {
+      this.insert_dropzone($("input[type='file']:visible"))
+
+      // initialize intersection observer to observe intersection of input node and viewport
+      // handler called when target is 0% visible
+      // https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API
+      const targetNodes = $("input[type='file']")
+      const intersectionHandler = (e) => {
+        console.log("OBSERVED")
+        if ($("input[type='file']:visible").length >= 1) {
+          console.log($("input[type='file']:visible"))
+          this.update_dropzone_input($("input[type='file']:visible"))
+        }
+      }
+
+      const observerHidden = new IntersectionObserver(intersectionHandler, { threshold: 0.0 })
+      targetNodes.each (function() {
+        observerHidden.observe(this)
+      })
```

Comment:
Instead of looking for changes to the page shape/scroll, can we just look for a drag-over event?  This works better, but not quite correctly on pages like this: https://hmis-warehouse.dev.test/data_sources/82/uploads/new.  There are two or three file inputs, but some are hidden behind collapsible items.  It seems to correctly pick one if only one is visible at a time, but if multiple are visible, it just puts the file on the first.
If we can add a listener for when a file is dragged over the page, I think the only change to the previous code needed would be to limit to `:visible`.